src/compas_ags/rhino/forceobject.py

```diff
@@ -70,8 +61,8 @@ def scale_from_3_points(self):
         gp.SetCommandPrompt('Select the 2nd reference point.')
 
         def OnDynamicDraw(sender, e):
-            base_ref1 = base.DistanceTo(ref1)
-            base_ref2 = base.DistanceTo(e.CurrentPoint)
+            base_ref1 = Point3d(* origin).DistanceTo(ref1)
```

Comment:
constantly converting the origin to `Point3d` seems unnecessary. please convert once at the start and then reuse...

Reply:
also, is `base_ref1` not just `L1` and `base_ref2` `L2`?

Reply:
or `d1` and `d2`